include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,519 @@
+         * beginning of the program. Checks preconditions.
+         */
+        virtual
+        void
+        initialize ();
+
+        /**
+         * A function that is called at the beginning of each time step. For
+         * the current plugin, this function loads the next data files if
+         * necessary and outputs a warning if the end of the set of data
+         * files is reached.
+         */
+        virtual
+        void
+        update ();
+
+        /**
+         * Return the boundary velocity as a function of position. For the
+         * current class, this function returns value from the text files.
+         */
+        Tensor<1,dim>
+        boundary_velocity (const Point<dim> &position) const;
+
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * A variable that stores the currently used data file of a
+         * series. It gets updated if necessary by set_current_time.
+         */
+        int  current_file_number;
+
+        /**
+         * Time from which on the data file with number 'First data
+         * file number' is used as boundary condition. Previous to this
+         * time, a no-slip boundary condition is assumed. Depending on the setting
+         * of the global 'Use years in output instead of seconds' flag
+         * in the input file, this number is either interpreted as seconds or as years."
+         */
+        double first_data_file_model_time;
+
+        /**
+         * Number of the first data file to be loaded when the model time
+         * is larger than 'First data file model time'.
+         */
+        int first_data_file_number;
+
+        /**
+         * In some cases the boundary files are not numbered in increasing
+         * but in decreasing order (e.g. 'Ma BP'). If this flag is set to
+         * 'True' the plugin will first load the file with the number
+         * 'First data file number' and decrease the file number during
+         * the model run.
+         */
+        bool decreasing_file_order;
+
+        /**
+         * Directory in which the data files are present.
+         */
+        std::string data_directory;
+
+        /**
+         * Filename of data file. The file names can contain
+         * the specifiers %s and/or %c (in this order), meaning the name of the
+         * boundary and the number of the data file time step.
+         */
+        std::string data_file_name;
+
+        /**
+         * Boundary id for which this plugin is created. Is initialized in
+         * initialize() and used to determine the appropriate file name.
+         */
+        types::boundary_id boundary_id;
```

Comment:
It would seem like all of these member variables here are the same as in the various other plugins, or do I see this wrong? If so, then this would suggest that they all be derived from some common base class that provides the member variables and the functions that operate on them.